### PR TITLE
[receiver/mysql] Add mysql.connection.errors metric

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Add mysql.connection_errors metric

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14138

**Testing:**
unit tests

**Documentation:**
`metadata.yaml`

### CODE REVIEW

File: receiver/mysqlreceiver/metadata.yaml [hunk 1]
```diff
@@ -279,3 +283,13 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [threads]
+  mysql.connection_errors:
```
Comment (djaglowski) [inline receiver/mysqlreceiver/metadata.yaml:286]: It seems like a `mysql.connection` namespace could be useful for other metrics such as `.count`, which would be related to this metric. What do you think about calling this `mysql.connection.errors`?

Comment (sumo-drosiek) [reply receiver/mysqlreceiver/metadata.yaml]: sure, done

**Review by djaglowski — APPROVED** [review]

**Comment by djaglowski** [comment]
LGTM, but need to resolve the conflict

**Comment by djaglowski** [comment]
Closed based on #15366

**Pull request closed without merging**